src/Datadog.Trace/TracingProcessManager.cs

```diff
@@ -359,8 +365,20 @@ public void InitializePortFileWatcher()
                 _portFileWatcher.EnableRaisingEvents = true;
             }
 
+            private static string ReadSingleLineNoLock(string file)
+            {
+                using (var fs = new FileStream(file, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                {
```

Comment:
After researching whether this will work when the file is already being read/written elsewhere, I see that using `FileShare.ReadWrite` is necessary. Nice!